sopel/modules/wikipedia.py

```diff
@@ -161,14 +163,13 @@ def mw_snippet(server, query):
 def say_section(bot, trigger, server, query, section):
     page_name = query.replace('_', ' ')
     query = quote(query.replace(' ', '_'))
-    section = section.replace('_', ' ')
 
     snippet = mw_section(server, query, section)
     if not snippet:
         bot.say("[WIKIPEDIA] Error fetching section \"{}\" for page \"{}\".".format(section, page_name))
         return
 
-    msg = '[WIKIPEDIA] {} - {} | "{}"'.format(page_name, section, snippet)
+    msg = '[WIKIPEDIA] {} - {} | "{}"'.format(page_name, section.replace('_', ' '), snippet)
```

Comment:
Had a couple thoughts while looking at this before bed. Cursory stuff, not a formal review (and not that this PR needs another until it's "really ready"), but things that might inform the evolution as it's getting close to done.

I wonder how many users would be confused if that separating `-` turned into a `§`… Somehow the hyphen doesn't seem right. Section symbol, some Unicode arrow, `>`… one of those maybe?

Also: What do you think about returning a tuple from `mw_section()`, say `(section_title, snippet)`? Have to think about the possibility of HTML, character entities, etc. in the section title, though. The `replace` here is certainly the simplest way to go!

Reply:
I find the hyphen to fit better than the alternatives, but that's wholly a personal opinion.

Wrt. the tuple suggestion, I presume you're referring to taking the entry from the MediaWiki sections listing. That'd involve at least some HTML parsing in some cases (replacing `<i></i>` with a call to `formatting.italic()`, e.g.) which I'd remove from my implementation, but could be done.

Reply:
I don't think parsing HTML in the section title is worth it. Stripping tags is fine. It's just that the anchor name is sanitized of more than just HTML tags, and might be significantly different from the "real" section title that `mw_section()` has access to.

`§` seems appropriate because that's literally the reason that symbol exists—to denote a document section—but I'm not going to push. It can always be changed later. :smile_cat: